### PR TITLE
Add wallet button

### PR DESCRIPTION
**Pull request by jellyfish-tom**
## What does this PR do?

Adds ProgressButton that later on can be used in Airdrop wallets list:
image

#### *Any scout cleaning/refactoring?*
removed unused React imports

## Todos:

- [X] Checked on iOS
- [ ] Checked on Android

### CODE REVIEW

File: src/components/ProgressButton.tsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+}: Props) => {
+  const [inProgress, setInProgress] = useState(false);
+  const [progress, setProgress] = useState(0);
+  const [intervalID, setIntervalID] = useState(noIntervalID);
+
+  const addWallet = () => {
+    setInProgress(true);
+
+    const intervalID = setInterval(() => {
+      setIntervalID((intervalID as unknown) as number);
```
Comment (jellyfish-tom) [inline src/components/ProgressButton.tsx:33]: couldn't find type to use here (or actually what to pass to `useState(` in line 27). When I log `typeof intervalID` in line 32 it s `number`. I should probably use `setInterval` from object, not straight 'from air' like I do. Checked documentation but couldn't find good answer. 
Any suggestions :)?

File: src/components/ProgressButton.tsx [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+  stepIntervalMilis,
+  onComplete,
+  title,
+  inProgressTitle,
+}: Props) => {
+  const [inProgress, setInProgress] = useState(false);
+  const [progress, setProgress] = useState(0);
+  const [intervalID, setIntervalID] = useState(noIntervalID);
+
+  const addWallet = () => {
```
Comment (KedziaPawel) [inline src/components/ProgressButton.tsx:29]: this button shouldn't be aware of a wallet, it should be a generic dump component

File: src/components/ProgressButton.tsx [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+  stepIntervalMilis: number;
+  onComplete: () => any;
+  title: string;
+  inProgressTitle: string;
+}
+
+const noIntervalID = -1;
+
+export const ProgressButton: FC<Props> = ({
+  timeoutMilis,
```
Comment (KedziaPawel) [inline src/components/ProgressButton.tsx:19]: settingInterval on every rerender could end badly for the memory consumption and would end up probably with a memory leak. I've read the documentation and I think that setting `indeterminateAnimationDuration` to the `duration` taken from props and setting in the useEffect(() => {
 setInProgress(1)
}, [])
could work
Zrzut ekranu 2021-01-19 o 15 32 19

Comment (KedziaPawel) [reply src/components/ProgressButton.tsx:19]: sorry, there is no set interval on every rerender, my bad:) anyway,  please try the suggested solution if it works

Comment (jellyfish-tom) [reply src/components/ProgressButton.tsx:19]: yeah, there shouldn't be any memory leaks and seems to work just fine. Refactor anyway?

File: src/components/ProgressButton.tsx (hunk 2 above)
Comment (jellyfish-tom) [reply src/components/ProgressButton.tsx]: done

File: src/components/ProgressButton.tsx (hunk 3 above)
Comment (KedziaPawel) [reply src/components/ProgressButton.tsx:19]: yes please, seems to be a simpler solution, no need to add custom logic if the component already has props for that. 
There could be a memory leak if the component would unmount before setInterval ends. To make it work properly there should be clearInterval on the unmounting

Comment (jellyfish-tom) [reply src/components/ProgressButton.tsx:19]: when using indeterminate it looks like this (so we can't use it):
image

Comment (KedziaPawel) [reply src/components/ProgressButton.tsx:19]: ok, so please add clearInterval on the unmounting

File: src/components/ProgressButton.tsx (hunk 1 above)
Comment (KedziaPawel) [reply src/components/ProgressButton.tsx:37]: Zrzut ekranu 2021-01-19 o 16 15 10
no TS error in my console

File: src/components/ProgressButton.tsx (hunk 3 above)
Comment (jellyfish-tom) [reply src/components/ProgressButton.tsx:19]: I've added clearInterval as a cleanup function in return from useEffect

File: src/components/ProgressButton.tsx (hunk 1 above)
Comment (marek-siemieniuk-morawski) [reply src/components/ProgressButton.tsx:39]: Maybe let's rename `intervalID` in the state or inside the `onPress` function? There are no conflicts or syntax errors as far as I can see but I'm not sure it's a good idea to have two variables with the same name.

**Review by marek-siemieniuk-morawski — APPROVED** [review]

File: decs.d.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+declare module 'react-native-progress/Bar';
```
Comment (KedziaPawel) [inline decs.d.ts:1]: we have src/types.d.ts for that

**Review by KedziaPawel — APPROVED** [review]

File: decs.d.ts (hunk 4 above)
Comment (jellyfish-tom) [reply decs.d.ts:1]: done

File: src/components/ProgressButton.tsx (hunk 1 above)
Comment (jellyfish-tom) [reply src/components/ProgressButton.tsx:39]: done

**Pull request merged**